src/main/java/com/suse/saltstack/netapi/client/SaltStackJdkConnection.java

```diff
@@ -0,0 +1,110 @@
+package com.suse.saltstack.netapi.client;
+
+import com.suse.saltstack.netapi.config.SaltStackClientConfig;
+import com.suse.saltstack.netapi.exception.SaltStackException;
+import com.suse.saltstack.netapi.utils.SaltStackClientUtils;
+
+import com.google.gson.Gson;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+
+/**
+ * Class representation of a connection to SaltStack for issuing API requests using JDK's HttpURLConnection.
+ */
+public class SaltStackJdkConnection implements SaltStackConnection {
```

Comment:
I know that it doesn't really matter, but for 3 letter acronyms (that cannot be [avoided] (http://www.oracle.com/technetwork/java/codeconventions-135099.html)) I would prefer to put them all uppercase in class names. So consider to rename this class (and related ones accordingly) to `SaltStackJDKConnection` or maybe get rid of JDK in the class names overall.